src/rgw/rgw_op.cc

```diff
@@ -3514,3 +3531,335 @@ void RGWHandler::put_op(RGWOp *op)
   delete op;
 }
 
+
+
+/* Object Rename operation */
+
+int RGWRenameObj::verify_permission()
+{
+    // This function used to check ACLs in legacy code
```

Comment:
Shouldn't this function have something like-:
 if (!verify_bucket_permission(s, RGW_PERM_WRITE))
    return -EACCES;